code/modules/mob/living/carbon/carbon.dm

```diff
@@ -714,7 +714,7 @@
 
 	//Fire and Brute damage overlay (BSSR)
 	var/hurtdamage = getBruteLoss() + getFireLoss() + damageoverlaytemp
-	if(hurtdamage)
+	if(hurtdamage && !HAS_TRAIT(src, TRAIT_NO_PAIN))
```

Comment:
Could you `if(HAS_TRAIT(src, TRAIT_NO_PAIN)) return` above this line and simply never apply the brute overlay in the first place?